android/app/src/main/java/io/locha/p2p/runtime/ChatService.java

```diff
@@ -70,15 +74,23 @@ public void setEventsHandler(ChatServiceEvents eventsHandler) {
      *
      * @throws RuntimeException if the server is already started.
      */
-    public void start(byte[] privateKey) {
+    public void start(byte[] privateKey, Promise promise) {
```

Comment:
Could we use the Promise only on `ChatServiceModule.java` also with a try-catch block? this to report errors to JavaScript, as some exceptions can occur when API is misused.